pkg/utils/test/object.go

```diff
@@ -0,0 +1,119 @@
+package test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http/httptest"
+	"net/url"
+	"reflect"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+type hookErrorCheck func(context.Context) error
+
+// ObjectTests contains the logic to test any Object implementation with the hooks it implements, checking
+// * if the Object actually implements the interface of the hook
+// * if the Object has an identifier field
+// * calling the hook function with incomplete contexts gives none or the correct error (meaning the error is handled correctly)
+//
+func ObjectTests(o types.Object, hooks ...interface{}) {
+	ginkgo.It("has an identifier", func() {
+		_, err := api.GetObjectIdentifier(o, false)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	})
+
+	for _, hook := range hooks {
+		name := reflect.TypeOf(hook).Elem().Name()
```

Comment:
I would move this line a few lines down. To initialize the variable shortly before it is used. makes it a little bit more readable

Reply:
ack